### ISSUE TITLE
install

### ISSUE THREAD

**Issue by luislopes7** [c3d/tao-3D#3]
hello,I am trying to install tao3d in linux without success, I downloaded it from ´´https://sourceforge.net/projects/tao3d/´´ but when I run''make install'' I get this error:
Make [1]: *** [compiler.o] Error 1
Make [1]: exits the directory "/ home / nada / tao-3D / libxlr"
Make: *** [sub-libxlr-install_subtargets] Error 2
The truth is that there is not too much information about this language and its installation even on its website, I ask if you can guide me since I am a bit lost.
Thank you

**Comment by c3d** [comment]
Hello, I am sorry for the bad experience.

There is insufficient data in the messages you copied. Apparently, there is a compile error compiling "compiler.c", but you have not copy-pasted it above. Could you please copy the complete transcript? Also, to help me reproduce, can you tell me which version of Linux? I'll install it in a VM and try to do the same.

And yes, I know that I need to return spending more time on these projects, but I've been extremely busy with the DxO ONE lately, and did not have much time for my other projects. Sorry again.

Are you the same person that asked about installation on Grenouille Bouillie? If so, I think that github may be a better place and I will redirect people from my blog here.

Thank you for letting me know about the problem.

**Comment by luislopes7** [comment]
Yes, I am the same person, the linux version is ubuntu 14.04 lts 32bits
Qt version 4.8.6.
Git version 1.9.1
 G ++ 4.8.4
Do not check the version of llvm but I'm almost sure it's 2.9
I attach the output to the terminal.

[tao3d.txt](https://github.com/c3d/tao-3D/files/627555/tao3d.txt)

**Comment by c3d** [comment]
According to the build log, you are building with LLVM 3.4 and not 2.9. While XL (the language underlying Tao3D), should support that if I remember correctly, I don't know if the version used by Tao3D. I did not have time to take a look at it this week-end (there was a relatively large change in the ELFE "flight recorder" (https://github.com/c3d/elfe/commit/695a80568651cd4e63ba95c4c959555d30bfce43), which will make it much easier to debug some thorny issues in Tao3D.

If you want to know why supporting several versions of LLVM are problematic, I invite you to read this file:  https://github.com/c3d/elfe/blob/master/src/llvm-crap.h :-)

**Comment by luislopes7** [comment]
Thank you man!! for the concern, I do not pretend that you waste your time answering my silly questions, answer me as much as you can.
In spite of everything I would like to try to program something with tao. I build llvm 2.9 but now I get this error:
nada@nada:~/tao-3D$ ./configure
Running: qmake -r CONFIG+=release CONFIG+=silent PACKAGE_APPEND=   
Reading /home/nada/tao-3D/libxlr/libxlr.pro
Project ERROR: Can't execute llvm-config-2.9 nor llvm-config

**Comment by c3d** [comment]
You probably do not have the llvm-config in your PATH. Try to find where it was installed, and add that to the PATH variable before running qmake.

Christophe

**Comment by luislopes7** [comment]
I could pass ./configure successfully
But now I have this error.

nada@nada:~/tao-3D$ make install
cd libxlr/ && make -f Makefile install
make[1]: se ingresa al directorio «/home/nada/tao-3D/libxlr»
rm -f libxlr.so.1.0.0 libxlr.so libxlr.so.1 libxlr.so.1.0
linking libxlr.so.1.0.0
/usr/bin/ld: no se puede encontrar -lLLVMObject
/usr/bin/ld: no se puede encontrar -lLLVMMCJIT
/usr/bin/ld: no se puede encontrar -lLLVMMCDisassembler
/usr/bin/ld: no se puede encontrar -lLLVMLinker
/usr/bin/ld: no se puede encontrar -lLLVMJIT
/usr/bin/ld: no se puede encontrar -lLLVMMCParser
/usr/bin/ld: no se puede encontrar -lLLVMMC
collect2: error: ld returned 1 exit status
make[1]: *** [libxlr.so.1.0.0] Error 1
make[1]: se sale del directorio «/home/nada/tao-3D/libxlr»
make: *** [sub-libxlr-install_subtargets] Error 2

**Comment by c3d** [comment]
OK will try to replicate this when I’m back at home (I’m currently travelling). Expect a response around next Monday. If not, please ping me back to put some pressure, something like “if you don’t make this work today, I kill a kitten”.

Thanks for reporting.
Christophe

**Comment by luislopes7** [comment]
I think it solves the problem of libraries, but it still throws me errors when I execute make, in the txt files are the output:
[configure_out.txt](https://github.com/c3d/tao-3D/files/660086/configure_out.txt)
[make_install_out.txt](https://github.com/c3d/tao-3D/files/660087/make_install_out.txt)

**Comment by luislopes7** [comment]
the kitten is dead Christophe

**Comment by c3d** [comment]
Hi Luis,

My Linux installations did not go as smoothly as planned, see https://grenouillebouillie.wordpress.com/2016/12/21/goodbye-dxo-hello-red-hat. So I must admit that I did not spend much time on Tao3D yet. And then there was this little thing called Christmas. But still planning to get back to it.

Christophe

> On 27 Dec 2016, at 23:20, luislopes7 <notifications@github.com> wrote:
> 
> the kitten is dead Christophe
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

Christophe de Dinechin

**Comment by c3d** [comment]
Hi Luis. It took me a while, but I believe that the kitten is back on its feet. Please let me know if you have better luck now.

**Comment by luislopes7** [comment]
No,i dont have luck, same problem to build

**Comment by luislopes7** [comment]
did you see the output on txt files?

**Comment by c3d** [comment]
Yes. I thought this specific issue was fixed (the "cannot stat"). So I re-opened the issue.

BTW, when you send output files like this, would you mind temporarily setting LANG=C, this would make it easier for me to figure out what the messages mean.

**Comment by c3d** [comment]
I will install an Ubuntu 14.04 32-bit VM to see if I can replicate this.

**Comment by revoker** [comment]
Hi there.

I saw today the [talk about XL and Tao at fosdem 2020](https://fosdem.org/2020/schedule/event/xllang/), I know this repos is not that active but in order to improve on the current state, I've tried to make a containerized build. Even if it's not yet successful, I guess this improves a little bit...

So on a linux box with docker installed, clone the repos and its submodules
```sh
git clone https://github.com/c3d/tao-3D.git
cd tao-3D
git submodule update --init --recursive
```

After that create a [Dockerfile.txt (from the attachement)](https://github.com/c3d/tao3D/files/4954607/Dockerfile.txt)  and try:
```sh
mv Dockerfile.txt Dockerfile  # Github disallow the attachement without extension

# Launch the build
docker build .
```
The dockerfile contains a version of llvm2.9 (along with llvm3.4, hence the rename) with all the prerequisites (qt5, opengl).

Although `./configure --llvm-config /usr/local/llvm/llvm-2.9/bin/llvm-config` detects the llvm with the right version, I couldn't pass the `make install` as it fails with this message


```log
cd libxlr/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /usr/src/myapp/libxlr/libxlr.pro CONFIG+=release CONFIG+=silent LLVMCONFIG=/usr/local/llvm/llvm-2.9/bin/llvm-config PACKAGE_APPEND= DEFINES+=CFG_NO_QTWEBKIT NO_HELP_VIEWER=1 -o Makefile ) && make -f Makefile install
make[1]: Entering directory `/usr/src/myapp/libxlr'
compiling xlr/xlr/basics.cpp
<command-line>:0:1: error: macro names must be identifiers
[...]
In file included from xlr/xlr/compiler.h:42:0,
                 from xlr/xlr/symbols.h:48,
                 from xlr/xlr/basics.cpp:50:
xlr/xlr/llvm-crap.h: In constructor 'LLVMCrap::JIT::JIT()':
xlr/xlr/llvm-crap.h:378:37: error: 'InitializeNativeTargetAsmParser' was not declared in this scope
     InitializeNativeTargetAsmParser();
                                     ^
xlr/xlr/llvm-crap.h:379:40: error: 'InitializeNativeTargetDisassembler' was not declared in this scope
     InitializeNativeTargetDisassembler();
[...]
```

The full [build log is here ](https://github.com/c3d/tao3D/files/4954728/build.log)

Hope this helps.

**Comment by c3d** [comment]
Thanks for the report, @revoker . I'm trying this right now. I added the `--verbose` option to the `configure` step. I suspect part of the problem is the output of `llvm-config`
